api/src/main/java/org/openmrs/module/locationbasedaccess/PatientSearchAdvise.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.locationbasedaccess;
+
+
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInvocation;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.PersonAttribute;
+import org.openmrs.api.context.Context;
+import org.springframework.aop.Advisor;
+import org.springframework.aop.support.StaticMethodMatcherPointcutAdvisor;
+import java.lang.reflect.Method;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * It will used to implement the AOP Methods for PatientServiceImpl class
+ */
+public class PatientSearchAdvise extends StaticMethodMatcherPointcutAdvisor implements Advisor {
+
+    protected static final Log log = LogFactory.getLog(PatientSearchAdvise.class);
+
+    @Override
+    public boolean matches(Method method, Class targetClass) {
+        // For PatientServiceImpl.savePatient()
+        if (method.getName().equals("getPatients")) {
+            return true;
+        }
+        // For PatientServiceImpl.getPatient()
+        else if (method.getName().equals("getPatient")) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Advice getAdvice() {
+        return new UserControlAdvice();
+    }
+
+    /**
+     * This inner class will be used to intercept the caught method using AOP
+     */
+    private class UserControlAdvice implements MethodInterceptor {
```

Comment:
What are we using this class for?

Reply:
AOP Advice method which needs to be triggered. I have renamed this class according to the usage.